apollo-openapi/src/test/java/com/ctrip/framework/apollo/openapi/client/ApolloOpenApiClientIntegrationTest.java

```diff
@@ -0,0 +1,169 @@
+import com.ctrip.framework.apollo.openapi.dto.OpenReleaseDTO;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Do not run in 'mvn clean test',
+ * left code here for develop test,
+ * you can run the method by ide.
+ */
+class ApolloOpenApiClientIntegrationTest {
+
+  private final Logger log = LoggerFactory.getLogger(this.getClass());
+
+  private final ApolloOpenApiClient client = newClient();
+
+  private final String env = "DEV";
+  private final String clusterName = "default";
+
+  ApolloOpenApiClient newClient() {
+    String someUrl = "http://localhost:8070";
+//    String someToken = "0627b87948c30517157e8b2a9565e473b5a97323a50128f584838ed10559d3fd";
+    String someToken = "9d0a241e9cb2300f302a875b1195340b2b6f56373cf5ca5d006a3f4e1a46b3ef";
+
+    return ApolloOpenApiClient.newBuilder()
+        .withPortalUrl(someUrl)
+        .withToken(someToken)
+        .withReadTimeout(2000 * 1000)
+        .withConnectTimeout(2000 * 1000)
+        .build();
+  }
+
+  void createApp(String appId, String ownerName, String ... admins) {
+    {
+      OpenCreateAppDTO req = new OpenCreateAppDTO();
+      req.setName("openapi create app 测试名字 " + appId);
+      req.setAppId(appId);
+      req.setOwnerName(ownerName);
+      req.setOwnerEmail(ownerName + "@apollo.apollo");
+      req.setOrgId("orgIdFromOpenapi");
+      req.setOrgName("orgNameFromOpenapi");
+      req.setAdmins(new HashSet<>(Arrays.asList(admins)));
+      client.createApp(req);
+    }
+  }
+
+  @Test
+  @Disabled("only for integration test")
+  public void testCreateApp() {
+    final String appId = "openapi-create-app1";
+    final String ownerName = "user-test-xxx1";
+    createApp(appId, ownerName, "user-test-xxx2", "user3");
+
+    List<OpenAppDTO> list = client.getAppsByIds(Collections.singletonList(appId));
+    assertEquals(1, list.size());
+    OpenAppDTO openAppDTO = list.get(0);
+    log.info("{}", openAppDTO);
+    assertEquals(appId, openAppDTO.getAppId());
+    assertEquals(ownerName, openAppDTO.getOwnerName());
+  }
+
+  @Test
+  @Disabled("only for integration test")
+  public void testCreateAppThenCreateNamespaceThenRelease() {
+    // create app
+    final String appId = "openapi-create-app1";
+    final String ownerName = "test-create-release1";
+    createApp(appId, ownerName, "user-test-xxx1", "user-test-xxx2");
+
+    {
+      List<OpenAppDTO> list = client.getAppsByIds(Collections.singletonList(appId));
+      assertEquals(1, list.size());
+      OpenAppDTO openAppDTO = list.get(0);
+      log.info("{}", openAppDTO);
+      assertEquals(appId, openAppDTO.getAppId());
+      assertEquals(ownerName, openAppDTO.getOwnerName());
+    }
+
+    // create namespace
```

Comment:
Should we allow user create namespame, modify the item, and release it? i.e 

![image](https://github.com/apolloconfig/apollo-java/assets/15523186/d73dd806-010a-4987-8806-48b315e17087)

If a openapi token create the appId, it own the appId too.

I think it will give user a better experience.

For the code, we can introduce an boolean mark(grantAppPermission) in OpenCreateAppDTO?

Reply:
I believe this situation is indeed valid.